src/server/svr_jobfunc.c

```diff
@@ -714,6 +713,8 @@ svr_setjobstate(job *pjob, int newstate, int newsubstate)
 			}
 			/* if subjob, update parent Array Job */
 			if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob) {
+				if (JOB_STATE_FINISHED == newstate)
```

Comment:
Update a comment explaining why we had to do this.

Reply:
Since subjob never go to F state (instead go to X) , the if block is added to guard if any one setting it to F state.

Reply:
Can you update the same in comments. That helps people reading the code why it is done.

Reply:
Added in rc 13 https://github.com/PBSPro/pbspro/pull/590/commits/82b8f337b566ec556777394e4e3ef00fc07a0d4b